src/System.Private.Uri/src/System/UriSyntax.cs

```diff
@@ -202,22 +195,22 @@ internal bool IsAllSet(UriSyntaxFlags flags)
         private bool IsFullMatch(UriSyntaxFlags flags, UriSyntaxFlags expected)
         {
             // Return true, if masking the current set of flags with 'flags' equals 'expected'.
-            // Definition 'current set of flags': 
+            // Definition 'current set of flags':
             // a) if updatable flags were never set: m_Flags
             // b) if updatable flags were set: set union between all flags in m_Flags which are not updatable
             //    (i.e. not part of c_UpdatableFlags) and all flags in m_UpdatableFlags
 
             UriSyntaxFlags mergedFlags;
 
             // if none of the flags in 'flags' is an updatable flag, we ignore m_UpdatableFlags
-            if (((flags & c_UpdatableFlags) == 0) || !_updatableFlagsUsed)
```

Comment:
This looks broken to me.  If _updatableFlagsUsed is being deleted because it was never used, then it was always false, in which case !_updatableFlagsUsed was always true, which means this condition would have always been true (since anything ||'d with true is true, but now with this change it could be false.

Reply:
The Uri tests are failing now in CI.  So, clearly this is not dead code.  This change should be reverted.